pkg/cloud/vsphere/provisioner/common/templates.go

```diff
@@ -561,6 +570,70 @@ nodeRegistration:
     key: node-role.kubernetes.io/master
 EOF
 
+{{- end }}{{/* end MajorVersion 1.11 or 1.12 */}}
+
+{{ if eq .MajorVersion "1.13" }}
+cat > /etc/kubernetes/kubeadm_config.yaml <<EOF
+apiVersion: kubeadm.k8s.io/v1beta1
+kind: InitConfiguration
+nodeRegistration:
+  criSocket: /var/run/dockershim.sock
```

Comment:
if they are defaults from the kubeadm side, then in the template here should be just capture the fields that we do want to override, and not specify the others (as far as possible)?
That way, if there are changes to kubeadm in the future versions, we don't explicitly set and override the defaults, rather have kubeadm use whatever the defaults it has. What do you think?

Reply:
I think it's out-of-scope for this PR.  That is, I think that the way the kubeadm script is templated should be expanded greatly, and this configuration should be revisited with every quarterly kubernetes release. How much can we separate the kubeadm configuration from the cluster-api code entirely?

Reply:
oh, BTW in 1.14 we are going to have auto-detection for CRIs so this has to be omitted unless you really want to hard code docker in the provider.